hail/python/hail/table.py

```diff
@@ -2334,6 +2333,8 @@ def join(self,
         while the key fields from the right table are not present in
         the result.
 
+        Missing (NA) keys never match.
```

Comment:
This may be mentioned elsewhere, but if not, we should describe NA != NA; this can be confusing.